src/test/java/pl/wojciechkostecki/revel/controller/MenuControllerIT.java

```diff
@@ -0,0 +1,223 @@
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON))
+                .andDo(MockMvcResultHandlers.print())
+                .andExpect(MockMvcResultMatchers.status().isCreated());
+
+        MenuDTO menu2 = new MenuDTO();
+        menu2.setName("Menu");
+        menu2.setLocalId(local.getId());
+
+        int dbSize = menuRepository.findAll().size();
+
+        //when
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/menus")
+                .content(objectMapper.writeValueAsString(menu2))
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON))
+                .andDo(MockMvcResultHandlers.print())
+                .andExpect(MockMvcResultMatchers.status().isBadRequest())
+                .andExpect(result -> Assertions.assertTrue(result.getResolvedException() instanceof BadRequestException))
+                .andExpect(result -> Assertions.assertEquals("Menu is already assigned to local",
+                        result.getResolvedException().getMessage()));
+        ;
+
+        //then
+        int dbSizeAfter = menuRepository.findAll().size();
+
+        assertThat(dbSizeAfter).isEqualTo(dbSize);
+    }
+
+    @Test
+    void createMenuWhenTheLocalCannotBeFoundTest() throws Exception {
+        //given
+        MenuDTO menu = new MenuDTO();
+        menu.setName("Menu Stary Browar");
+        menu.setLocalId(3L);
+
+        int dbSize = menuRepository.findAll().size();
+
+        //when
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/menus")
+                .content(objectMapper.writeValueAsString(menu))
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON))
+                .andDo(MockMvcResultHandlers.print())
+                .andExpect(MockMvcResultMatchers.status().isNotFound())
+                .andExpect(result -> Assertions.assertTrue(result.getResolvedException() instanceof EntityNotFoundException))
+                .andExpect(result -> Assertions.assertEquals("Couldn't find a local with passed id: " + menu.getLocalId(),
+                        result.getResolvedException().getMessage()));
+
+
+        //then
+        int dbSizeAfter = menuRepository.findAll().size();
+
+        assertThat(dbSizeAfter).isEqualTo(dbSize);
+    }
+
+    @Test
+    void getAllMenus() throws Exception {
+        //given
+        Menu menu = new Menu();
+        menu.setName("Menu Bue Bue");
+        menuRepository.save(menu);
+
+        Menu menu2 = new Menu();
+        menu2.setName("Menu Stary Browar");
+        menuRepository.save(menu2);
+
+        //when
+        MvcResult mvcResult = mockMvc.perform(MockMvcRequestBuilders.get("/api/menus"))
+                .andDo(MockMvcResultHandlers.print())
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andReturn();
+
+        //then
+        List<Menu> menus = objectMapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<List<Menu>>() {});
```

Comment:
thats wrong because you are returning dto from the endpoint if i remember correctly, but dto doesnt have `equals` method.